apps/prairielearn/src/ee/lib/validateHTML.ts

```diff
@@ -0,0 +1,616 @@
+    case 'pl-number-input':
+      return checkNum(ast);
+    case 'pl-answer':
+      return checkMCAnswer(ast);
+    case 'pl-overlay':
+      return checkOverlay(ast);
+    case 'pl-location':
+      return checkLocation(ast);
+    case 'pl-order-blocks':
+      return checkOrderBlocks(ast);
+    case 'pl-matrix-input':
+      return checkMatrix(ast);
+  }
+  return [];
+}
+
+/**
+ * Checks that a multiple choice question tag has valid properties.
+ * @param ast The tree to consider, rooted at the tag.
+ * @returns The list of errors for the tag, if any.
+ */
+function checkMC(ast: any): string[] {
+  const errors: string[] = [];
+  let usedAnswersName = false;
+  let displayDropdown = false;
+  let usedAllOfTheAbove = false;
+  let usedNoneOfTheAbove = false;
+  let usedAllOfTheAboveFeedback = false;
+  let usedNoneOfTheAboveFeedback = false;
+  let usedSize = false;
+  const optionsOfTheAbove = ['false', 'random', 'correct', 'incorrect'];
+  for (const attr of ast.attrs) {
+    const key = attr.name;
+    const val = attr.value;
+    switch (key) {
+      case 'answers-name':
+        usedAnswersName = true;
+        break;
+      case 'weight':
+        assertInt('pl-multiple-choice', key, val, errors);
+        break;
+      case 'display':
+        assertInChoices('pl-multiple-choice', key, val, ['block', 'inline', 'dropdown'], errors);
+        if (val === 'dropdown') {
+          displayDropdown = true;
+        }
+        break;
+      case 'number-answers':
+        assertInt('pl-multiple-choice', key, val, errors);
+        break;
+      case 'order':
+        assertInChoices(
+          'pl-multiple-choice',
+          key,
+          val,
+          ['random', 'ascend', 'descend', 'fixed'],
+          errors,
+        );
+        break;
+      case 'hide-letter-keys':
+        assertBool('pl-multiple-choice', key, val, errors);
+        break;
+      case 'all-of-the-above':
+        assertInChoices('pl-multiple-choice', key, val, optionsOfTheAbove, errors);
+        if (optionsOfTheAbove.includes(val) && val !== 'false') {
+          usedAllOfTheAbove = true;
+        }
+        break;
+      case 'none-of-the-above':
+        assertInChoices('pl-multiple-choice', key, val, optionsOfTheAbove, errors);
+        if (optionsOfTheAbove.includes(val) && val !== 'false') {
+          usedNoneOfTheAbove = true;
+        }
+        break;
+      case 'all-of-the-above-feedback':
+        usedAllOfTheAboveFeedback = true;
+        break;
+      case 'none-of-the-above-feedback':
+        usedNoneOfTheAboveFeedback = true;
+        break;
+      case 'allow-blank':
+        assertBool('pl-multiple-choice', key, val, errors);
+        break;
+      case 'size':
+        usedSize = true;
+        assertInt('pl-multiple-choice', key, val, errors);
+        break;
+      default:
+        if (!HTML_GLOBAL_TAGS.includes(key)) {
+          errors.push(`${key} is not a valid property for pl-multiple-choice.`);
+        }
```

Comment:
Why are these being special cased? When are these ever valid for use as an attribute on a PL element?

Reply:
Changed in 0e87e7e4333c2d4d8dfbc2116aad0c0acd81707b. Some generations were given `id` tags, so I kept it for now.